Feature/142 housing available read all endpoint

VitaRox opened the pull request:
## **In progress:**
Making this route use 'jwt' auth stratetgy. Tested and works as expected (auth'd users can view all listed Housing Availables;

VitaRox committed:
- Adds basic implementation of GET (all) route in routes and controllers;
- Doesn't quite work yet, though returns the expected error message if no Housing Availables are found;

VitaRox committed:
- Appears to display all Housing Availables that have been listed in preliminary Swagger tests!;

VitaRox committed:
- Slight stylistic corrections; more tests run and verified to return all previously-entered HousingAvailables;

VitaRox committed:
- Adds in the ability for auth'd users to view all listed Housing Availables;

zaclittleberry commented on lib/housing-available/housing-available.routes.js:
```diff
@@ -1,3 +1,5 @@
+const hapiAuthJwt2 = require("hapi-auth-jwt2");
+
```
no need to import/require this here, please remove. the auth strategy declaration/config happens in index.js. All we need to do is pass the string 'jwt' to auth (which you've done) and hapi will look up the strategy we've already registered,

zaclittleberry commented on lib/housing-available/housing-available.routes.js:
```diff
@@ -17,6 +19,20 @@ module.exports = {
+          tags: ['api', 'HousingAvailable'],
+          validate: {
+            query: housingAvailableModels.apiFilterQuery,
+          }
```
Doesn't look like any filters are configured in the handler, so this should be removed until the functionality is there.

zaclittleberry reviewed (changes requested):
small changes. just removing a few things that don't need to be there. This is the ticket we tested together in our meeting, right?

Commit:
Update housing-available.routes.js

Removes 2 unnecessary bits of code;

Commit:
Update housing-available.controllers.js

Fixes error codes to be more uniform; fixes a curly brace/scope issue.

zaclittleberry commented on lib/housing-available/housing-available.controllers.js:
```diff
@@ -182,17 +182,26 @@ module.exports = (models) => {
-
       } catch (e) {
-        throw Boom.badImplementation("Housing available instance creation failed.");
+        throw Boom.badImplementation('Error during createHousingAvailable(housingAvailableObject): ' + e);
```
please pass e as a second argument to .badImplementation. it should be `'...housingAvailableObject).', e`

zaclittleberry commented on lib/housing-available/housing-available.controllers.js:
```diff
@@ -182,17 +182,26 @@ module.exports = (models) => {
+      try {
+        response = await models.HousingAvailable.findAll();
+      } catch (e) {
+        throw Boom.badImplementation('Error during models.HousingAvailable.findAll(). ' + e);
```
e as second arg

Commit:
Update housing-available.controllers.js

- Fixes issue in lines 186 and 197, passing e as a 2nd argument instead of concatenating it to the prior string.

zaclittleberry reviewed (approved):
Approved!
